src/Workspaces/Core/Portable/Workspace/Solution/ChecksumCollection.cs

```diff
@@ -2,81 +2,109 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
+using System;
 using System.Collections;
 using System.Collections.Generic;
 using System.Collections.Immutable;
-using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.PooledObjects;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.Serialization
+namespace Microsoft.CodeAnalysis.Serialization;
+
+/// <summary>
+/// A wrapper around an array of <see cref="Microsoft.CodeAnalysis.Checksum"/>s, which also combines the value into a
+/// single aggregate checksum exposed through <see cref="Checksum"/>.
+/// </summary>
+internal sealed class ChecksumCollection(ImmutableArray<Checksum> children) : IReadOnlyCollection<Checksum>, IChecksummedObject
```

Comment:
>(ImmutableArray<Checksum> children) [](http://example.com/codeflow?start=40&length=35)

Wanted to see if you could help me understand the reasoning behind the naming of the primary ctor params. Maybe I just need to change how I think about primary ctors, but I think of "children" as a member of this class, and thus it should be "_children". Pretty much every time I see a reference to this inside a method body, I end up having to do a little mind bending to figure out it's a member, and not a local.

Reply:
Mind hasn't yet been warped to accept this as truth, let's chat in person.